### PR TITLE
nginx: on encountering lua directives, produce a better warning

### PR DESCRIPTION
**Pull request by alexzorin**
See #9066.

---

The code is a little bit layer violating but it seems a bit tricky to do this from `configurator.py`, given that we are swallowing the parsing errors.

The output becomes is as below. It's unfortunate that the warning is printed twice, once on presenting the challenge response, and one on cleaning it up. But: this already happens on `master`, the situation is fairly rare and still achieves a big improvement with telling the user what's wrong.

image

The difference compared to `master` is:

image

### CODE REVIEW

File: certbot-nginx/tests/testdata/etc_nginx/unsupported_directives.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+server {
+    listen 80;
+
+    location / {
+        foobar_by_lua_block {
+            ngx.say("Hello World")
+        }
+    }
+}
```
Comment (ohemorange) [inline certbot-nginx/tests/testdata/etc_nginx/unsupported_directives.conf:11]: missing newline

**Review by ohemorange — APPROVED** [review]
This looks great, helpful solution to get a better error message out there, thanks!

**Comment by ohemorange** [comment]
Haven't merged it because I wasn't sure if you wanted to add a changelog entry.

**Comment by alexzorin** [comment]
Thank you!

**Pull request merged**